yarn-project/telemetry-client/src/start.ts

```diff
@@ -7,13 +7,27 @@ import { type TelemetryClient } from './telemetry.js';
 
 export * from './config.js';
 
-export async function createAndStartTelemetryClient(config: TelemetryClientConfig): Promise<TelemetryClient> {
+let initialised = false;
+let telemetry: TelemetryClient = new NoopTelemetryClient();
+
+export async function initTelemetryClient(config: TelemetryClientConfig): Promise<TelemetryClient> {
   const log = createLogger('telemetry:client');
+  if (initialised) {
+    log.warn('Telemetry client has already been initialized once');
+    return telemetry;
+  }
+
+  initialised = true;
   if (config.metricsCollectorUrl) {
     log.info('Using OpenTelemetry client');
-    return await OpenTelemetryClient.createAndStart(config, log);
+    telemetry = await OpenTelemetryClient.createAndStart(config, log);
   } else {
     log.info('Using NoopTelemetryClient');
-    return new NoopTelemetryClient();
   }
+
+  return telemetry;
+}
+
+export function getTelemetryClient(): TelemetryClient {
+  return telemetry;
```

Comment:
This means that if a component accidentally asks for a telemetry client before the init call has completed, it'll get a noop telemetry client instead. I'd vote for making this call async and merging get and init, so we know we always get the client we expect.

Reply:
Actually, we should be able to init the client synchronously. The only reason `OpenTelemetryClient.createAndStart` is async is because of an async resource, but otel already handles async resources under the hood, so we don't really need the `await waitForAsyncAttributes` during initialization I believe.

So we should be able to:
- Drop the `await waitForAsyncAttributes` in the otel resource
- Make otel client initialization synchronous
- Have a single sync `getTelemetryClient` that initializes if needed, and delete `initTelemetryClient`

Reply:
> Have a single sync getTelemetryClient that initializes if needed, and delete initTelemetryClient

Telemetry config can come from --cmd-line-flags too https://github.com/AztecProtocol/aztec-packages/blob/2507b6fd1e1c79e7f27a2d3c357155670634aad5/yarn-project/aztec/src/cli/aztec_start_options.ts#L231

Maybe something to consider removing?

Agree on all of the other points and will update this PR.

Reply:
> This means that if a component accidentally asks for a telemetry client before

Good point and that's why I kept the client behind a getter instead of making it an export :grimacing: